File: CSharpCore/HamaraBasketTests/QualityCheckerTests.cs
using HamaraBasketQualityAssurance;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace HamaraBasketTests
{
    [TestClass]
    public class QualityCheckerTests
    {
        [TestMethod]
        public void QualityShouldDecreaseByOne()
        {
            var item = new HamaraBasketItem(10, 10, ItemType.Expiring);
            var checker = new QualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(9, item.Quality);
        }

        [TestMethod]
        public void QualityShouldDecreaseTwiceWhenSellByPassed()
        {
            var item = new HamaraBasketItem(0, 10, ItemType.Expiring);
            var checker = new QualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(8, item.Quality);
        }

        [TestMethod]
        public void QualityShouldDecreaseByOneWhenSellByIsZeroAndQualtiyIsOne()
        {
            var item = new HamaraBasketItem(0, 1, ItemType.Expiring);
            var checker = new QualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(0, item.Quality);
        }

        [TestMethod]
        public void QualityShouldNotBeNegative()
        {
            var item = new HamaraBasketItem(1, 0, ItemType.Expiring);
            var checker = new QualityChecker();

            checker.UpdateQuality(item);

            Assert.AreEqual(0, item.Quality);
        }
    }
}

File: Java/src/test/java/com/hamaraBasket/QualityCheckerTests.java
package com.hamaraBasket;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class QualityCheckerTests
    {
        @Test
        public void qualityShouldDecreaseByOne()
        {
            HamaraBasketItem item = new HamaraBasketItem(10, 10, ItemType.Expiring);
            QualityChecker checker = new QualityChecker();

            checker.updateQuality(item);

            assertEquals(9, item.getQuality());
        }

        @Test
        public void qualityShouldDecreaseTwiceWhenSellByPassed()
        {
            HamaraBasketItem item = new HamaraBasketItem(0, 10, ItemType.Expiring);
            QualityChecker checker = new QualityChecker();

            checker.updateQuality(item);

            assertEquals(8, item.getQuality());
        }

        @Test
        public void qualityShouldDecreaseByOneWhenSellByIsZeroAndQualtiyIsOne()
        {
            HamaraBasketItem item = new HamaraBasketItem(0, 1, ItemType.Expiring);
            QualityChecker checker = new QualityChecker();

            checker.updateQuality(item);

            assertEquals(0, item.getQuality());
        }

        @Test
        public void qualityShouldNotBeNegative()
        {
            HamaraBasketItem item = new HamaraBasketItem(1, 0, ItemType.Expiring);
            QualityChecker checker = new QualityChecker();

            checker.updateQuality(item);

            assertEquals(0, item.getQuality());
        }
    }